### PR TITLE
Add CompactionSpec.

### PR DESCRIPTION
**Pull request by toregge**
Pass CompactionStrategy to DataStoreBase::startCompactWorstBuffers in
preparation for compacting more buffers in the same compaction round.

@geirst : please review
@baldersheim : FYI

### CODE REVIEW

File: vespalib/src/vespa/vespalib/datastore/compaction_spec.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+{
+    bool _compact_memory;
+    bool _compact_address_space;
+public:
+    CompactionSpec(bool compact_memory, bool compact_address_space)
+        : _compact_memory(compact_memory),
+          _compact_address_space(compact_address_space)
+    {
+    }
+    bool get_compact_memory() const noexcept { return _compact_memory; }
```
Comment (geirst) [inline vespalib/src/vespa/vespalib/datastore/compaction_spec.h:24]: As this function returns a bool, I think it reads better when removing the _get__ prefix. Applies to the function below as well.

**Review by geirst — APPROVED** [review]
👍

File: vespalib/src/vespa/vespalib/datastore/compaction_spec.h (hunk 1 above)
Comment (toregge) [reply vespalib/src/vespa/vespalib/datastore/compaction_spec.h]: Fixed.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**